From 847019fc75b3534da1a97bd101a4ce800f42792e Mon Sep 17 00:00:00 2001
From: Kenneth Feng <kenneth.feng@amd.com>
Date: Fri, 27 Sep 2024 16:48:06 +0800
Subject: [PATCH 1/3] drm/amd/pm: update the default power limit on smu 13.0.0

update the default power limit on smu 13.0.0 for better performance

Link: https://gitlab.freedesktop.org/drm/amd/-/issues/3618
Link: https://gitlab.freedesktop.org/drm/amd/-/issues/3131
Fixes: c50fe289ed7207 ("drm/amdgpu/swsmu: always force a state reprogram on init")
Signed-off-by: Kenneth Feng <kenneth.feng@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/pm/swsmu/smu13/smu_v13_0_0_ppt.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/pm/swsmu/smu13/smu_v13_0_0_ppt.c b/drivers/gpu/drm/amd/pm/swsmu/smu13/smu_v13_0_0_ppt.c
index 1d024b122..a0dc891d9 100644
--- a/drivers/gpu/drm/amd/pm/swsmu/smu13/smu_v13_0_0_ppt.c
+++ b/drivers/gpu/drm/amd/pm/swsmu/smu13/smu_v13_0_0_ppt.c
@@ -2363,11 +2363,12 @@ static int smu_v13_0_0_get_power_limit(struct smu_context *smu,
 		power_limit = smu->adev->pm.ac_power ?
 			      skutable->SocketPowerLimitAc[PPT_THROTTLER_PPT0] :
 			      skutable->SocketPowerLimitDc[PPT_THROTTLER_PPT0];
-
+	if (power_limit < msg_limit)
+		smu_v13_0_set_power_limit(smu, SMU_DEFAULT_PPT_LIMIT, msg_limit);
 	if (current_power_limit)
-		*current_power_limit = power_limit;
+		*current_power_limit = msg_limit;
 	if (default_power_limit)
-		*default_power_limit = power_limit;
+		*default_power_limit = msg_limit;
 
 	if (powerplay_table) {
 		if (smu->od_enabled &&
-- 
2.46.0.469.g4590f2e941

